src/NHibernate/SqlCommand/SqlString.cs

```diff
@@ -347,7 +347,23 @@ public SqlString Append(string text)
 		{
 			if (string.IsNullOrEmpty(text)) return this;
 			if (_length == 0) return new SqlString(text);
-			return new SqlString(new object[] { this, text });
+			return new SqlString(this, text);
+		}
+
+		public SqlString Append(params object[] parts)
```

Comment:
I would consider introducing a new `private SqlString(SqlString other, params object[] parts)` constructor which can optimise the parts copying and append operations.

Reply:
Actually I've considered such ctor. But looking at existing ctors `SqlString(SqlString x)` and `SqlString(params object[] parts)` - I didn't find a way to merge them nicely. So IMHO it's not worth it. But if you've come up with implementation  - feel free to add it.